src/Tizen.NUI/src/public/Capture.cs

```diff
@@ -0,0 +1,252 @@
+                {
+                    finishedCallback = onFinished;
+                    finishedSignal.Disconnect(finishedCallback);
+                }
+            }
+        }
+
+        /// <summary>
+        /// CaptureFinishedState
+        /// </summary>
+        public enum CaptureFinishedState
+        {
+            /// <summary>
+            /// Succeeded
+            /// </summary>
+            Succeeded = 0,
+            /// <summary>
+            /// Failed
+            /// </summary>
+            Failed = 1,
+        }
+
+        /// <summary>
+        /// GetNativeImageSource
+        /// </summary>
+        /// <returns></returns>
+        public NativeImageSource GetNativeImageSource()
+        {
+            Tizen.Log.Debug("NUI", $"GetNativeImageSource()");
+            return new NativeImageSource(Interop.Capture.GetNativeImageSourcePtr(swigCPtr), true);
+        }
+
+    }
+
+    /// <summary>
+    /// CaptureFinishStateEventArgs
+    /// </summary>
+    public class CaptureFinishStateEventArgs : EventArgs
+    {
+        public Capture.CaptureFinishedState State { get; internal set; }
+    }
+
+
+    internal class CaptureSignal : Disposable
+    {
+        internal CaptureSignal(IntPtr cPtr, bool cMemoryOwn) : base(cPtr, cMemoryOwn)
+        {
+        }
+
+        protected override void ReleaseSwigCPtr(System.Runtime.InteropServices.HandleRef swigCPtr)
+        {
+            if (swigCMemOwn)
+            {
+                Interop.Capture.DeleteSignal(swigCPtr);
+            }
+        }
+
+        public bool Empty()
+        {
+            bool ret = Interop.Capture.SignalEmpty(swigCPtr);
+            if (NDalicPINVOKE.SWIGPendingException.Pending) throw NDalicPINVOKE.SWIGPendingException.Retrieve();
+            return ret;
+        }
+
+        public uint GetConnectionCount()
+        {
+            uint ret = Interop.Capture.SignalGetConnectionCount(swigCPtr);
+            if (NDalicPINVOKE.SWIGPendingException.Pending) throw NDalicPINVOKE.SWIGPendingException.Retrieve();
+            return ret;
+        }
+
+        public void Connect(Delegate func)
+        {
+            IntPtr ip = Marshal.GetFunctionPointerForDelegate<Delegate>(func);
+            {
+                Interop.Capture.SignalConnect(swigCPtr, new HandleRef(this, ip));
+                if (NDalicPINVOKE.SWIGPendingException.Pending) throw NDalicPINVOKE.SWIGPendingException.Retrieve();
+            }
+        }
+
+        public void Disconnect(Delegate func)
+        {
+            IntPtr ip = Marshal.GetFunctionPointerForDelegate<Delegate>(func);
+            {
+                Interop.Capture.SignalDisconnect(swigCPtr, new HandleRef(this, ip));
+                if (NDalicPINVOKE.SWIGPendingException.Pending) throw NDalicPINVOKE.SWIGPendingException.Retrieve();
+            }
+        }
+
+        public void Emit(Capture src, Capture.CaptureFinishedState state)
```

Comment:
The parameter can be like bool CaptureSuccess, and this bool value is needed to changed to the enum to be used in the dali SignalEmit.